PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/ui/WebAppDeployMvpView.java

```diff
@@ -40,7 +40,7 @@
 
 public interface WebAppDeployMvpView extends MvpView {
 
-    void renderWebAppsTable(@NotNull List<ResourceEx<WebApp>> webAppLists);
+    void renderWebAppsAndSlots(@NotNull List<ResourceEx<WebApp>> webAppLists);
```

Comment:
Why do we have to change the interface?

Reply:
The logic here is that: the deploy slot panel depends on the web apps table is ready. Enabling the deploy slot panel without any web apps ready, or even no web apps in the selected subscription is weird for users. 

So when the presenter finishes loading web apps, it will render the web app table and the slot panel in this one function call. `renderWebAppsAndSlots` describes more clear what the function in charge of.